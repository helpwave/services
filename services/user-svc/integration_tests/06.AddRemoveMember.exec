#!/bin/bash

grpcurl --plaintext -H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "d311e7d9-85cd-4ae9-a0c0-7a0abc6dd48e", "user_id": "9b045845-b5bf-456b-94a3-0b7bfd0bd0c5"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/AddMember 2>&1 || error ""

echo "get members:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiI0NDYxMmM1YS00ZTYxLTQzMTgtOGYyYi1iNzM0ZWVmMzA1ZDIiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{ "id":"d311e7d9-85cd-4ae9-a0c0-7a0abc6dd48e" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetMembersByOrganization

echo "remove member:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{ "id":"d311e7d9-85cd-4ae9-a0c0-7a0abc6dd48e", "user_id": "9b045845-b5bf-456b-94a3-0b7bfd0bd0c5" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/RemoveMember 2>&1 || error ""

echo "get members:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiI0NDYxMmM1YS00ZTYxLTQzMTgtOGYyYi1iNzM0ZWVmMzA1ZDIiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{ "id":"d311e7d9-85cd-4ae9-a0c0-7a0abc6dd48e" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetMembersByOrganization
