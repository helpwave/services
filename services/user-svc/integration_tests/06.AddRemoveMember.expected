{}
get members:
{
  "members": [
    {
      "userId": "9b045845-b5bf-456b-94a3-0b7bfd0bd0c5",
      "email": "annika.b@helpwave.de",
      "nickname": "Annika",
      "avatarUrl": "https://source.boringavatars.com/marble/128/"
    },
    {
      "userId": "44612c5a-4e61-4318-8f2b-b734eef305d2",
      "email": "member.tester@helpwave.de",
      "nickname": "mem",
      "avatarUrl": "https://source.boringavatars.com/marble/128/"
    }
  ]
}
remove member:
{}
get members:
{
  "members": [
    {
      "userId": "44612c5a-4e61-4318-8f2b-b734eef305d2",
      "email": "member.tester@helpwave.de",
      "nickname": "mem",
      "avatarUrl": "https://source.boringavatars.com/marble/128/"
    }
  ]
}
